pkg/provider/provider.go

```diff
@@ -98,6 +98,15 @@ type KeyVaultObject struct {
 	// The encoding of the object in KeyVault
 	// Supported encodings are Base64, Hex, Utf-8
 	ObjectEncoding string `json:"objectEncoding" yaml:"objectEncoding"`
+	// Permission is the file permissions
+	Permission string `json:"permission" yaml:"permission"`
+}
+
+// SecretFile holds content and metadata of a secret file
+type SecretFile struct {
+	Content  []byte `json:"content" yaml:"content"`
+	Path     string `json:"path" yaml:"path"`
+	FileMode string `json:"fileMode" yaml:"fileMode"`
```

Comment:
The fields in this struct aren't used for encoding. So we can drop the tags here?

Reply:
yup. I was trying different approach for octal conversion which I eventually dropped and forgot to remove this. Now removed.